<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>801</width>
    <height>566</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>761</width>
      <height>521</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_4">
        <attribute name="title">
         <string>IP</string>
        </attribute>
        <widget class="QWidget" name="gridLayoutWidget_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>741</width>
           <height>417</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="14" column="1">
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Fragment offset</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_28">
            <property name="text">
             <string>Type of Service(8 bits)</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Internet Header Length</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_21">
            <property name="text">
             <string>Length</string>
            </property>
           </widget>
          </item>
          <item row="13" column="0">
           <widget class="QLabel" name="label_26">
            <property name="text">
             <string>Data</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QLineEdit" name="lineEdit_IPProt"/>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_23">
            <property name="text">
             <string>Checksum</string>
            </property>
           </widget>
          </item>
          <item row="12" column="1">
           <widget class="QLineEdit" name="lineEdit_IPOpt"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_27">
            <property name="text">
             <string>Version</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="lineEdit_IPLen"/>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit_IPIHL"/>
          </item>
          <item row="12" column="0">
           <widget class="QLabel" name="label_29">
            <property name="text">
             <string>Options</string>
            </property>
           </widget>
          </item>
          <item row="13" column="1">
           <widget class="QPlainTextEdit" name="plainTextEdit_IPData"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_24">
            <property name="text">
             <string>Identificator</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLineEdit" name="lineEdit_IPFrag"/>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_25">
            <property name="text">
             <string>Protocol &lt;----</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QLineEdit" name="lineEdit_IPTTL"/>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="lineEdit_IPTOS"/>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="label_30">
            <property name="text">
             <string>Destination IP</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>Flags(3 bits)</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QLineEdit" name="lineEdit_IPChecksum"/>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="lineEdit_IPID"/>
          </item>
          <item row="5" column="1">
           <widget class="QLineEdit" name="lineEdit_IPFlags"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>TTL</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEdit_IPVersion"/>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="label_31">
            <property name="text">
             <string>Source IP</string>
            </property>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QLineEdit" name="lineEdit_IPSrc"/>
          </item>
          <item row="11" column="1">
           <widget class="QLineEdit" name="lineEdit_IPDst"/>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_5">
        <attribute name="title">
         <string>UDP</string>
        </attribute>
        <widget class="QWidget" name="gridLayoutWidget">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>9</y>
           <width>741</width>
           <height>351</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Cheksum</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="lineEdit_UDPLen"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Source Port</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit_UDPDst"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Data</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Length</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Destination Port</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="lineEdit_UDPChecksum"/>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEdit_UDPSrc"/>
          </item>
          <item row="4" column="1">
           <widget class="QPlainTextEdit" name="plainTextEdit_UDPData"/>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_6">
        <attribute name="title">
         <string>TCP</string>
        </attribute>
        <widget class="QWidget" name="gridLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>741</width>
           <height>365</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="7" column="0">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Window size</string>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Options</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Sequence number</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>Urgent pointer</string>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Checksum</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Data offset</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Destination Port</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>Flags(9 bits)</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Acknowlegment number</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Source Port</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Reserved</string>
            </property>
           </widget>
          </item>
          <item row="12" column="1">
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Data</string>
            </property>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QLineEdit" name="lineEdit_TCPOptions"/>
          </item>
          <item row="10" column="2">
           <widget class="QLineEdit" name="lineEdit_TCPOptionValue"/>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPSrc"/>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPDst"/>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPSeq"/>
          </item>
          <item row="3" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPAck"/>
          </item>
          <item row="4" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPDataOffset"/>
          </item>
          <item row="5" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPReserved"/>
          </item>
          <item row="6" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPFlags"/>
          </item>
          <item row="7" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPWindow"/>
          </item>
          <item row="8" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPChecksum"/>
          </item>
          <item row="9" column="1" colspan="2">
           <widget class="QLineEdit" name="lineEdit_TCPUrgent"/>
          </item>
          <item row="11" column="1" colspan="2">
           <widget class="QPlainTextEdit" name="plainTextEdit_TCPData"/>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_7">
        <attribute name="title">
         <string>ICMP</string>
        </attribute>
        <widget class="QWidget" name="gridLayoutWidget_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>741</width>
           <height>351</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="label_33">
            <property name="text">
             <string>Type</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_34">
            <property name="text">
             <string>Data</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_32">
            <property name="text">
             <string>Identificator</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="lineEdit_ICMPChecksum"/>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit_ICMPCode"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_35">
            <property name="text">
             <string>Checksum</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="lineEdit_ICMPID"/>
          </item>
          <item row="5" column="1">
           <widget class="QPlainTextEdit" name="plainTextEdit_ICMPData"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_36">
            <property name="text">
             <string>Code</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEdit_ICMPType"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_37">
            <property name="text">
             <string>Sequence number</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="lineEdit_ICMPSeq"/>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_8">
        <attribute name="title">
         <string>Queue</string>
        </attribute>
        <widget class="QTableWidget" name="tableWidget_Queue">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>741</width>
           <height>421</height>
          </rect>
         </property>
         <column>
          <property name="text">
           <string>Protocol</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>IP_Src</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>IP_Dst</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Port_Src</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Port_Dst</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>ICMP_Type</string>
          </property>
         </column>
        </widget>
       </widget>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="pushButton_create">
         <property name="text">
          <string>Create</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_send">
         <property name="text">
          <string>Send</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_choose_int">
         <property name="text">
          <string>Choose interface</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox_interface"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>801</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
